packages/core-data/src/queried-data/selectors.js

```diff
@@ -79,7 +88,10 @@ function getQueriedItemsUncached( state, query ) {
 		} else {
 			// If expecting a complete item, validate that completeness, or
 			// otherwise abort.
-			if ( ! state.itemIsComplete[ itemId ] ) {
+			if (
+				! state.itemIsComplete[ context ] ||
+				! state.itemIsComplete[ context ][ itemId ]
```

Comment:
I guess you can use optional chaining here: `! state.itemIsComplete[ context ]?.[ itemId ]`.

Similar with the above `! state.items[ context ].hasOwnProperty( itemId )`.